scripts/tests/linux/tv_casting_test_sequences.py

```diff
@@ -0,0 +1,183 @@
+    - `output_msg` or `input_cmd` (mutually exclusive)
+
+    Note: The first entry in the test sequence list of steps should always be the output string to verify that the tv-app 
+    is up and running.
+
+    For output message blocks, define the start line, relevant lines, and the last line. If the last line contains trivial closing 
+    characters (e.g., closing brackets, braces, or commas), include the line before it with actual content. For example:
+        `Step(subprocess_='tv-casting-app', output_msg=['InvokeResponseMessage =', 'exampleData', 'InteractionModelRevision =', '},'])`
+
+    For input commands, define the command string with placeholders for variables that need to be updated. For example:
+        `Step(subprocess_='tv-casting-app', input_cmd='cast request 0\n')`
+    """
+
+    def __init__(
+        self,
+        app: App,
+        timeout_sec: Optional[int] = DEFAULT_TIMEOUT_SEC,
+        output_msg: Optional[List[str]] = None,
+        input_cmd: Optional[str] = None,
+    ):
+        # Validate that either `output_msg` or `input_cmd` is provided, but not both.
+        if output_msg is not None and input_cmd is not None:
+            logging.error('Step cannot contain both `output_msg` and `input_cmd`. Either `output_msg` or `input_cmd` should be provided.')
+            sys.exit(1)
+        elif output_msg is None and input_cmd is None:
+            logging.error('Step must contain either `output_msg` or `input_cmd`. Both are `None`.')
+            sys.exit(1)
+
+        # Define either `App.TV_APP` or `App.TV_CASTING_APP` on which we need to parse for `output_msg` or send `input_cmd`.
+        self.app = app
+
+        # Define the maximum time in seconds for timeout while parsing for the `output_msg`. If not provided, then we use the DEFAULT_TIMEOUT_SEC.
+        self.timeout_sec = timeout_sec
+
+        # Define the `output_msg` that we need to parse for in a list format.
+        self.output_msg = output_msg
+
+        # Define the `input_cmd` that we need to send to either the `App.TV_APP` or `App.TV_CASTING_APP`.
+        self.input_cmd = input_cmd
+
+
+class Sequence:
+    """A class representing a sequence of steps for testing the casting experience between the Linux tv-casting-app and the tv-app.
+
+    A Sequence object needs to be defined with an appropriate test sequence `name` along with its list of `Step` objects that will
+    be used for validating the casting experience.
+    """
+
+    def __init__(self, name: str, steps: List[Step]):
+        self.name = name
+        self.steps = steps
+
+
+def get_test_sequence_by_name(test_sequences: List[Sequence], test_sequence_name: str) -> Optional[Sequence]:
+    """Retrieve a test sequence from a list of sequences by its name."""
+
+    for sequence in test_sequences:
+        if sequence.name == test_sequence_name:
+            return sequence
+    return None
+
+
+def get_test_sequences() -> List[Sequence]:
```

Comment:
Is this a static method? Can we decorate this with @staticmethod. Same comment for get_test_sequence_by_name

Reply:
Addressed in the latest commit.